engine/xproviders/xproviders.go

```diff
@@ -105,21 +105,28 @@ func (pub *AdBuilder) BuildAndSign() (*schema.Advertisement, error) {
 	}
 
 	epMap := map[string]Info{}
-	for _, epInfo := range pub.providers {
-		adv.ExtendedProvider.Providers = append(adv.ExtendedProvider.Providers, schema.Provider{
-			ID:        epInfo.ID,
-			Addresses: epInfo.Addrs,
-			Metadata:  epInfo.Metadata,
-		})
-		epMap[epInfo.ID] = epInfo
-	}
+	if len(pub.providers) != 0 {
+		for _, epInfo := range pub.providers {
+			if len(epInfo.Addrs) == 0 || len(epInfo.Metadata) == 0 {
```

Comment:
* `nil` Metadata in extended provider is allowed right? which should pick up the root metadata if specified.
* Check for ID not being empty, by calling `.Validate()` on it.

Reply:
Good catch - done